libavutil/hwcontext_cuda.c

```diff
@@ -387,28 +387,36 @@ static int cuda_device_create(AVHWDeviceContext *device_ctx,
     if (device)
         device_idx = strtol(device, NULL, 0);
 
-    if (cuda_device_init(device_ctx) < 0)
+    if (cuda_device_init(device_ctx) < 0) {
+        ret = AVERROR_UNKNOWN; //return non retriable error
         goto error;
+    }
 
     cu = hwctx->internal->cuda_dl;
 
     ret = CHECK_CU(cu->cuInit(0));
-    if (ret < 0)
+    if (ret < 0) {
+        ret = AVERROR_UNKNOWN; //return non retriable error
         goto error;
+    }
 
     ret = CHECK_CU(cu->cuDeviceGet(&hwctx->internal->cuda_device, device_idx));
-    if (ret < 0)
+    if (ret < 0) {
+        ret = AVERROR(ENODEV); //return retriable error
```

Comment:
I suggested `ENXIO` as less widely-used code with a similar caption to reduce the chance of ambigious error messages. But good call on using `ENODEV`, it's probably fine.